core/deployment/src/main/java/io/quarkus/deployment/dev/IsolatedDevModeMain.java

```diff
@@ -210,6 +224,113 @@ private void startCodeGenWatcher(QuarkusClassLoader classLoader, List<CodeGenDat
+            // should not happen
+            return;
+        }
+        var arthasVersion = arthasVersionOpt.get();
+        var arthasDir = parentDir.resolve(String.format("arthas-%s", arthasVersion));
+        if (!Files.exists(arthasDir)) {
+            Files.createDirectory(arthasDir);
+        }
+
+        Runnable r;
+        if (!Files.exists(toArthasJar(arthasDir))) {
+            r = () -> startArthas(downloadArthas(arthasDir, arthasVersion), arthasHttpPort);
+        } else {
+            r = () -> startArthas(toArthasJar(arthasDir), arthasHttpPort);
+        }
+        var arthasThread = new Thread(r);
+        arthasThread.setName("Arthas thread");
+        arthasThread.setDaemon(true);
+        arthasThread.start();
+    }
+
+    private Path downloadArthas(Path arthasDir, String version) {
+        var downloadURL = "https://arthas.aliyun.com/download/" + version;
+        try {
+            log.debug("Attempting to download Arthas from " + downloadURL);
+            var arthasZipFile = File.createTempFile("arthas-" + version, ".zip");
+            try (BufferedInputStream in = new BufferedInputStream(new URL(downloadURL).openStream());
+                    FileOutputStream fileOutputStream = new FileOutputStream(arthasZipFile)) {
+                byte[] dataBuffer = new byte[10240];
+                int bytesRead;
+                while ((bytesRead = in.read(dataBuffer, 0, 10240)) != -1) {
+                    fileOutputStream.write(dataBuffer, 0, bytesRead);
+                }
+            }
+            ZipUtils.unzip(arthasZipFile.toPath(), arthasDir);
+            log.debug("Arthas downloaded");
+            return toArthasJar(arthasDir);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private void startArthas(Path arthasJar, Integer httpPort) {
+        log.debug("Attempting to start Arthas");
+        ProcessBuilder processBuilder = new ProcessBuilder("java",
+                "-jar", arthasJar.toAbsolutePath().toString(),
+                "" + ProcessHandle.current().pid(),
+                "--http-port",
+                "" + httpPort)
+                        .directory(arthasJar.getParent().toFile());
+
+        if (log.isDebugEnabled()) {
+            processBuilder.inheritIO();
+        } else {
+            processBuilder.redirectError(ProcessBuilder.Redirect.DISCARD.file())
+                    .redirectOutput(ProcessBuilder.Redirect.DISCARD.file());
+
+        }
+
+        try {
+            var arthasProcess = processBuilder.start();
+            try {
+                // attaching to the process takes a while
+                // TODO: we should probably tail the logs instead of waiting
+                Thread.sleep(5_000);
+            } catch (InterruptedException ignored) {
+
+            }
+            if (arthasProcess.isAlive()) {
+                log.infof("Arthas started and is accessible at http://127.0.0.1:%d", 8563);
```

Comment:
Maybe use localhost? I'm not sure the IPv4 IP works for someone configured with just IPv6.

Reply:
Just pushedthat change :)